### PR TITLE
Align CLC configuration with other agents + bump to latest releases

### PR DESCRIPTION
**Pull request by vboulineau**
#### What this PR does / why we need it:
Bump to latest Agent/Cluster Agent versions

#### Which issue this PR fixes
Fix issues with Cluster checks when tls_verify is required and endpoint checks are backed by podRef.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/datadog/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Datadog changelog
 
+## 2.3.11
+
+* Bump to Agent 7.20.2 and Cluster Agent 1.6.0
+* Align Cluster Check Runner environment configuration with other agent (fix issues with some endpoints checks backed by PODs)
```
Comment (ahmed-mez) [inline stable/datadog/CHANGELOG.md]: I think we can drop this line

Comment (vboulineau) [reply stable/datadog/CHANGELOG.md]: Rephrased it to be more accurate

**Pull request merged**